cpp/CMakeLists.txt

```diff
@@ -12,10 +13,43 @@ enable_testing()
 if(TOSCA_ASAN)
   add_compile_options(
     $<$<CXX_COMPILER_ID:Clang>:-fno-omit-frame-pointer>
-    $<$<CXX_COMPILER_ID:Clang>:-fsanitize=address>)
+    $<$<CXX_COMPILER_ID:Clang>:-fsanitize=address>
+    )
   add_link_options(
     $<$<CXX_COMPILER_ID:Clang>:-fno-omit-frame-pointer>
-    $<$<CXX_COMPILER_ID:Clang>:-fsanitize=address>)
+    $<$<CXX_COMPILER_ID:Clang>:-fsanitize=address>
+    )
+endif()
+
+if(TOSCA_COVERAGE)
+  if (NOT CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    # gcc uses gcov, clang should use it as well, but it has version compatibility issues
+    # between compiler generate code and the gcov tool.
+    message(FATAL_ERROR "Coverage is only supported with GCC.")
+  endif()
+
+  add_compile_options($<$<CXX_COMPILER_ID:GNU>:--coverage>)
+  add_link_options($<$<CXX_COMPILER_ID:GNU>:--coverage>)
+
+  find_program(LCOV lcov REQUIRED)
+  find_program(GENHTML genhtml REQUIRED)
+
+  add_custom_target(coverage
+    COMMENT "Generating coverage report."
+    COMMAND ${LCOV} 
+      --capture 
+      --directory .  
+      --ignore-errors mismatch 
```

Comment:
I got an error from this flag:
```
Generating coverage report.
Capturing coverage data from .
Subroutine read_intermediate_text redefined at /usr/bin/geninfo line 2623.
Subroutine read_intermediate_json redefined at /usr/bin/geninfo line 2655.
Subroutine intermediate_text_to_info redefined at /usr/bin/geninfo line 2703.
Subroutine intermediate_json_to_info redefined at /usr/bin/geninfo line 2792.
Subroutine get_output_fd redefined at /usr/bin/geninfo line 2872.
Subroutine print_gcov_warnings redefined at /usr/bin/geninfo line 2900.
Subroutine process_intermediate redefined at /usr/bin/geninfo line 2930.
Found gcov version: 11.4.0
Using intermediate gcov format
geninfo: ERROR: unknown argument for --ignore-errors: mismatch
gmake[4]: *** [CMakeFiles/coverage.dir/build.make:71: CMakeFiles/coverage] Error 255
gmake[4]: Leaving directory '/home/herbert/coding/fantom/tosca/cpp/build'
gmake[3]: *** [CMakeFiles/Makefile2:1048: CMakeFiles/coverage.dir/all] Error 2
gmake[3]: Leaving directory '/home/herbert/coding/fantom/tosca/cpp/build'
gmake[2]: *** [CMakeFiles/Makefile2:1055: CMakeFiles/coverage.dir/rule] Error 2
gmake[2]: Leaving directory '/home/herbert/coding/fantom/tosca/cpp/build'
gmake[1]: *** [Makefile:200: coverage] Error 2
gmake[1]: Leaving directory '/home/herbert/coding/fantom/tosca/cpp/build'
make: *** [Makefile:56: test-cpp-coverage] Error 2
```

I have LCOV version 1.14.

Reply:
ubuntu 20? 22? 
can you please run  `dpkg -S $(which lcov)`

Reply:
As expected, I am using the same Ubuntu version (22.04.) as @HerbertJordan and I get the same error.
Running the `dpkg` command return `/usr/bin/lcov`.

Reply:
Ok, problem comes from Gcc 11 vs 13:

1. gcc 11 does not have this ignore feature. This is default in ubuntu 22
2. gcc 13 complains about it and requires the argument. This is default in ubuntu 24

will see what can I do about it.

Reply:
should work now.